src/validators/object.ts

```diff
@@ -54,6 +54,70 @@ export class FieldsValidator<T> implements ValidatorInstance<Indexable<T>> {
   }
 }
 
+/**
+ * @api primitive
+ *
+ * The class that powers the `allFieldsPresent()` validator function.
+ */
+export class AllFieldsPresentValidator<T> implements ValidatorInstance<Indexable<T>> {
```

Comment:
Unless there is a use case for checking missing keys but not extra keys (or vice versa), I would suggest merging the two into a single `KeysValidator` that just checks for both.